code/game/objects/items/devices/instruments.dm

```diff
@@ -282,7 +282,7 @@
 		beacon_music(user)
 
 /obj/item/musicbeacon/proc/beacon_music(mob/M)
-	if(!display_names)
+	if(!display_names.len)

```

Comment:
Just make it not defined in type definition and only create the list in this block